FFWEB-839 & FFWEB-842 upgrading to v3 (#146)
FACT-Finder-Web-Components/documentation: markdown/3.0/en/documentation/upgrade-guide.md

## Upgrade from version 1.2.x to 3.0.0

> We strongly recommend to upgrade your system to use the latest version of FACT-Finder Web Components. While we will do critical bug fixes for older versions, new features are likely to be implemented only in newer versions. In addition the newer versions stick to newer technology decreasing your loading time and improving internal speed. We promise to make each upgrade progress as ease as possible.

Reviewer: ```suggestion
We strongly recommend upgrading your system to use the latest version of FACT-Finder Web Components. While we will do critical bug fixes for older versions, new features will likely only be implemented in newer versions. In addition, the newer versions utilize newer technology, decreasing your loading time and improving internal speed. We promise to make each upgrade go as smoothly as possible.
```

In general you can keep track of our progress, changes and new features in the [release notes](https://github.com/FACT-Finder-Web-Components/ff-web-components/blob/master/CHANGELOG.md).

> There are only two major changes and some minor API changes to take care of when upgrading from version 1.2.x to version 3.0.0. In addition css mixins have been removed. You can use normal css selector instead. Further more `ffw-` is introduced as name prefix for FACT-Finder Web Components custom class names.

Reviewer: ```suggestion
There are only two major changes and some minor API changes to take care of when upgrading from version 1.2.x to version 3.0.0. In addition, CSS mixins have been removed. You can use normal CSS selector instead. Further more `ffw-` is introduced as the prefix for FACT-Finder Web Components custom class names.
```

### 2 major changes

> - Aligning with the current trend FACT-Finder Web Components are now shipped as ES6 Module instead of HTML Imports. Hence you have to load `bundle.js` instead of the HTML import as follows:

Reviewer: ```suggestion
- Aligning with the current trend FACT-Finder Web Components are now shipped as ES6 Module instead of HTML Imports. Which is why you have to load `bundle.js` instead of the previous HTML import.
```

   
```html
   <!-- Before -->
   <script>
       var Polymer = Polymer || {};
       Polymer.dom = 'shady';
   </script>
   <script src="../bower_components/webcomponentsjs/webcomponents-lite.min.js"></script>
   <link rel="import" href="../bower_components/ff-web-components/dist/elements.build_with_dependencies.html">
   <style>
       [unresolved] {
           opacity: 0;
       }        
   </style>
     
   <!-- In version 3 -->
   <script src="../node_modules/ff-web-components/dist/vendor/custom-elements-es5-adapter.js"></script>
   <script src="../node_modules/ff-web-components/dist/vendor/webcomponents-loader.js"></script>
   <script defer src="../node_modules/ff-web-components/dist/bundle.js"></script>
 <style>
     [unresolved] {
         opacity: 0;
     }        
 </style>
```

- With [Polymer 3](https://www.polymer-project.org/3.0/docs/devguide/feature-overview) extending built-in HTML elements

> is not possible anymore. Hence you have to nest an `input` into `ff-searchbox` and a `button` into `ff-searchbutton`

Reviewer: ```suggestion
is not possible anymore. Instead you will have to put an `input` into `ff-searchbox` and a `button` into `ff-searchbutton`
```

as followed:

```html
    <!-- Before -->
    <ff-searchbox ...></ff-searchbox>
    <ff-searchbutton ...></ff-searchbutton>
           
    <!-- In version 3. -->
    <ff-searchbox ...>
        <input .../>
    </ff-searchbox>
    <ff-searchbutton>
        <button ...></button>
    </ff-searchbutton>
```
Note that native `input` properties like `placeholder` stay within the `input` tag, while enhancements like 

> `suggest-onfocus` move into `ff-searchbox`. If you had used css to style your search input and button don't forget to adjust your selectors accordingly.

Reviewer: ```suggestion
`suggest-onfocus` move into `ff-searchbox`. If you're using CSS to style your search input and button don't forget to adjust your selectors accordingly.
```

### Minor API changes to take care of

> Beside the two major changes the following list contains all breaking changes.

Reviewer: ```suggestion
The following list contains all remaining breaking changes.
```

> If we have missed something, we would be happy if you [contact](contacts) us.

Reviewer: ```suggestion
If we have missed something, we would be happy if you [contacted](contacts) us.
```
